test/integration/ssl_integration_test.h

```diff
@@ -29,6 +29,11 @@ class SslIntegrationTest : public HttpIntegrationTest,
   Network::ClientConnectionPtr makeSslClientConnection(bool alpn, bool san);
   void checkStats();
 
+protected:
+  // Set this true to debug SSL handshake issues with openssl s_client. The
+  // verbose trace will be in the logs, openssl must be installed separately.
+  bool debug_with_s_client_{false};
```

Comment:
Shall we use the `@borignssl//:bssl` cc_binary? Then this is self-contained.

Reply:
Yeah, it seems like it would be a better option, but it turns out that `bssl` is a bit impoverished when it comes to debug output. Take a look at this comparison: https://gist.github.com/htuch/a440a4edfaa059bc83647f3f3cd7083e (there are two files there, one for openssl with max debug, another for bssl). I think when chasing down very low level handshake issues in these tests, having the full `openssl s_client` output at disposal is the most helpful.

Reply:
Take a look at the gist I link to. openssl gives you a running commentary of different parts of the handshake, it dumps cert details and even the low level hex is sometimes interesting IMHO. Since this is a debug-only setting for one particular test, I'm hoping it's not too controversial :)

Reply:
I know what's the difference... What I'm asking is which of that extra information was actually helpful to you?

Reply:
Friendly ping, can we merge this one?

Reply:
Sorry for the delay, I'm OOO this week. Yeah I think it's fair to use `openssl`.